### PR TITLE
Moved requesting resources out of TaskComputer

### PR DESCRIPTION
**Pull request by Wiezzel**

### CODE REVIEW

File: golem/task/taskcomputer.py [hunk 1]
```diff
@@ -100,20 +100,10 @@ def __init__(
...
+    def task_given(self, ctd: 'ComputeTaskDef') -> None:
+        assert self.assigned_subtask is None
         self.assigned_subtask = ctd
-        self.__request_resource(
-            ctd['task_id'],
-            ctd['subtask_id'],
-            ctd['resources'],
-        )
-        return True
+        ProviderTimer.start()
```
Comment (Krigpl) [inline golem/task/taskcomputer.py:109]: That changes the semantics of what it measures and may not be wanted.

File: golem/task/taskcomputer.py [hunk 2]
```diff
@@ -366,9 +356,6 @@ def __request_task(self):
         if requested_task is not None:
             self.stats.increase_stat('tasks_requested')
 
-    def __request_resource(self, task_id, subtask_id, resources):
```
Comment (Krigpl) [inline golem/task/taskcomputer.py:369]: Removing `resource_failure` and `resource_collected` would also make sense to eliminate resource management completely from here, right?

File: golem/task/taskcomputer.py (hunk 1 above)
Comment (Wiezzel) [reply golem/task/taskcomputer.py:109]: How does it change the semantics? This statement was already there.

File: golem/task/taskcomputer.py (hunk 2 above)
Comment (Wiezzel) [reply golem/task/taskcomputer.py:369]: OK for `resource_failure` but `resource_collected` doesn't actually do any 'resource management'. It just starts computation. So maybe a rename could be appropriate.

File: golem/task/taskcomputer.py (hunk 1 above)
Comment (Krigpl) [reply golem/task/taskcomputer.py:109]: I meant that it used to include the time needed for the resource to be downloaded, now it doesn't.

Comment (Wiezzel) [reply golem/task/taskcomputer.py:109]: It still does. No logic has been changed here. Just moved to another class.

Comment (Krigpl) [reply golem/task/taskcomputer.py:109]: Derp, you're right, sorry.

**Comment by Wiezzel** [comment]
I refactored `resource_failure` and `resource_collected` as well.
Plx, review. @Krigpl @maaktweluit

File: golem/task/taskcomputer.py [hunk 3]
```diff
@@ -366,14 +343,20 @@ def __request_task(self):
...
+        docker_images = subtask['docker_images']
+        extra_data = subtask['extra_data']
+        subtask_deadline = subtask['deadline']
 
-    def __compute_task(self, subtask_id, docker_images,
-                       extra_data, subtask_deadline):
-        task_id = self.assigned_subtask['task_id']
         task_header = self.task_server.task_keeper.task_headers.get(task_id)
 
+        self.last_task_timeout_checking = time.time()
```
Comment (Krigpl) [inline golem/task/taskcomputer.py]: afaict this ain't no used can can be removed

**Review by Krigpl — APPROVED** [review]
thumbupglitch

File: golem/task/taskcomputer.py (hunk 3 above)
Comment (Wiezzel) [reply golem/task/taskcomputer.py]: Indeed.

File: golem/task/taskcomputer.py [hunk 4]
```diff
@@ -426,7 +404,9 @@ def __compute_task(self, subtask_id, docker_images,
         self.task_server.task_keeper.task_started(task_id)
         tt.start().addBoth(lambda _: self.task_computed(tt))
 
-    def __task_finished(self, ctd: 'ComputeTaskDef') -> None:
+    def _task_finished(self) -> None:
+        ctd = self.assigned_subtask
```
Comment (maaktweluit) [inline golem/task/taskcomputer.py:408]: maybe you can log a warning here when its already none

**Review by maaktweluit — APPROVED** [review]
LGTM!

One small comment, still approving :)

File: golem/task/taskcomputer.py (hunk 4 above)
Comment (Wiezzel) [reply golem/task/taskcomputer.py:408]: I'll add it my next PR because I don't want to start the CI again. :stuck_out_tongue:

**Pull request merged**